pkg/crd/markers/topology.go

```diff
@@ -110,4 +124,21 @@ func (m MapType) ApplyToSchema(schema *apiext.JSONSchemaProps) error {
 	return nil
 }
 
-func (l MapType) ApplyFirst() {}
+func (m MapType) ApplyFirst() {}
+
+func (s StructType) ApplyToSchema(schema *apiext.JSONSchemaProps) error {
+	if schema.Type != "object" {
```

Comment:
I suspect "object" is the default and is thus not required.

Reply:
Aha, makes sense. I [went for](https://github.com/kubernetes-sigs/controller-tools/blob/4752ed2de7d2fc1b6b18398bf26cf2ce6b53cd94/pkg/crd/markers/doc.go#L27-L29) `"object" || ""` just to be on the strict side. Let me know what you think, happy to remove the check completely if we think it doesn't add much.